### ISSUE TITLE
Test results across different versions

### ISSUE THREAD

**Issue by pfecht** [pltrdy/rouge#28]
I was trying to reproduce the scores with the following setup containing only the first sample from  your https://github.com/pltrdy/rouge/blob/master/tests/data.json:

*File 1: hyp.txt:* "the #### transcript is a written version of each day 's cnn student news program use this transcript to help students with reading comprehension and vocabulary use the weekly newsquiz to test your knowledge of storie s you saw on cnn student news"

*File 2 ref.txt:* "this page includes the show transcript use the transcript to help students with reading comprehension and vocabulary at the bottom of the page , comment for a chance to be mentioned on cnn student news . you must be a teacher or a student age # # or older to request a mention on the cnn student news roll call . the weekly newsquiz tests students ' knowledge of even ts in the news"

**1: Rouge v0.3.0:**

```bash
$ pip install rouge==0.3.0
$ rouge -f ./hyp.txt ./ref.txt
[
  {
    "rouge-1": {
      "f": 0.5411764657051903,
      "p": 0.6388888888888888,
      "r": 0.46938775510204084
    },
    "rouge-2": {
      "f": 0.27272726805123976,
      "p": 0.36585365853658536,
      "r": 0.21739130434782608
    },
    "rouge-l": {
      "f": 0.45001673716547075,
      "p": 0.5555555555555556,
      "r": 0.40816326530612246
    }
  }
]
```

**2: Rouge 0.2.0**

```bash
$ pip install rouge==0.2.0
$ rouge -f ./hyp.txt ./ref.txt
[
  {
    "rouge-1": {
      "f": 0.5348837160627367,
      "p": 0.6388888888888888,
      "r": 0.46
    },
    "rouge-2": {
      "f": 0.26785713821588014,
      "p": 0.36585365853658536,
      "r": 0.2112676056338028
    },
    "rouge-l": {
      "f": 0.28711800978275975,
      "p": 0.4418604651162791,
      "r": 0.25675675675675674
    }
  }
]
```
As you can see, the resuls for `rouge-l` are very different. However, the second results from version `0.2.0` equal the test data results of https://github.com/pltrdy/rouge/blob/master/tests/data.json. 

**3: Run with files2rouge from https://github.com/pltrdy/files2rouge** 
```bash
files2rouge hyp.txt ref.txt
Preparing documents...
Running ROUGE...
---------------------------------------------
1 ROUGE-1 Average_R: 0.69048 (95%-conf.int. 0.69048 - 0.69048)
1 ROUGE-1 Average_P: 0.42647 (95%-conf.int. 0.42647 - 0.42647)
1 ROUGE-1 Average_F: 0.52727 (95%-conf.int. 0.52727 - 0.52727)
---------------------------------------------
1 ROUGE-2 Average_R: 0.43902 (95%-conf.int. 0.43902 - 0.43902)
1 ROUGE-2 Average_P: 0.26866 (95%-conf.int. 0.26866 - 0.26866)
1 ROUGE-2 Average_F: 0.33333 (95%-conf.int. 0.33333 - 0.33333)
---------------------------------------------
1 ROUGE-L Average_R: 0.57143 (95%-conf.int. 0.57143 - 0.57143)
1 ROUGE-L Average_P: 0.35294 (95%-conf.int. 0.35294 - 0.35294)
1 ROUGE-L Average_F: 0.43636 (95%-conf.int. 0.43636 - 0.43636)
``` 
which is again different to the results in 1 and 2.

I know there are differences between the original perl implementation and the python reimplementation. But am I missing something or what is the reason for the very different results?

**Comment by pltrdy** [comment]
First, the most important difference between `0.2` and `0.3` ([see release] notes(https://github.com/pltrdy/rouge/releases)) is indeed about how ROUGE-L is calculated in the mult-sentence context. This fixes bad `0.2` behavior.

You show that there's small differences even considering ROUGE-1 & ROUGE-2, which I don't explain (should look closer, maybe not worth the investigation considering how tiny differences are). 

Now comparing `rouge` with `files2rouge` (which is in line with the official scores), there's important differences on data processing e.g. stemming, stopword removal etc. We cannot expect current python implementation to be super close to it for now it would require bit more stuff.

Also, ROUGE-L may still be incorrect, as it's discussed here: https://github.com/pltrdy/rouge/issues/27

Just to give background: the original idea of the repo was to get a fast pythonic implementation, that mimic ROUGE without being exactly it. For in-training evaluation or to use as a metric the exact value is not *that* important, you mostly want it to behave the same. For prediction evaluation or reproducing paper results -- which does not involve massive amount of scoring calls nor require high speed -- you better use `files2rouge`.

That being said, I'm really open to suggestions and PR (with scores and time evaluation).

**Comment by pfecht** [comment]
Thanks for the reply and the background on this repository. I missed the part about the 0.3 release so sorry about that.

Also many thanks for the distinction between in-training evaluation and actual paper results. That makes sense to me now.

I'll look closer into it to see if I can find any improvements. For now, this issue can be closed.
